dev/tools/update_versions.dart

```diff
@@ -0,0 +1,194 @@
+  argParser.addFlag(kHelp, negatable: false, help: 'Show this help message.');
+  final ArgResults argResults = argParser.parse(args);
+
+  final bool increment = argResults[kIncrement];
+  final bool brokeSdk = argResults[kBrokeSdk];
+  final bool brokeFramework = argResults[kBrokeFramework];
+  final bool brokeTest = argResults[kBrokeTest];
+  final bool brokeDriver = argResults[kBrokeDriver];
+  final bool brokeAnything = brokeSdk || brokeFramework || brokeTest || brokeDriver;
+  final bool release = argResults[kMarkRelease];
+  final bool help = argResults[kHelp];
+
+  if (help) {
+    print('update_versions.dart - update version numbers of Flutter packages and SDK');
+    print(argParser.usage);
+    exit(0);
+  }
+
+  if ((brokeAnything && release) || (brokeAnything && increment) || (release && increment)) {
+    print('You can either increment all the version numbers (--$kIncrement), indicate that some packages have had breaking changes (--broke-*), or switch to release mode (--$kMarkRelease).');
+    print('You cannot combine these, however.');
+    exit(1);
+  }
+
+  final RawVersion sdk = new RawVersion('VERSION');
+  final PubSpecVersion framework = new PubSpecVersion('packages/flutter/pubspec.yaml');
+  final PubSpecVersion test = new PubSpecVersion('packages/flutter_test/pubspec.yaml');
+  final PubSpecVersion driver = new PubSpecVersion('packages/flutter_driver/pubspec.yaml');
+
+  if (increment || brokeAnything)
+    sdk.increment(brokeAnything);
+  sdk.setMode(release);
+
+  if (increment || brokeFramework)
+    framework.increment(brokeFramework);
+  framework.setMode(release);
+
+  if (increment || brokeTest)
+    test.increment(brokeTest);
+  test.setMode(release);
+
+  if (increment || brokeDriver)
+    driver.increment(brokeDriver);
+  driver.setMode(release);
+
+  sdk.write();
+  framework.write();
+  test.write();
+  driver.write();
+
+  print('Flutter SDK is now at version: $sdk');
+  print('flutter package is now at version: $framework');
+  print('flutter_test package is now at version: $test');
+  print('flutter_driver package is now at version: $driver');
+}
+
+abstract class Version {
+  Version() {
+    read();
+  }
+
+  @protected
+  final List<int> version = <int>[];
+
+  @protected
+  bool dev;
+
+  @protected
+  bool dirty = false;
+
+  @protected
+  void read();
+
+  void interpret(String value) {
+    dev = value.endsWith(kDev);
+    if (dev)
+      value = value.substring(0, value.length - kDev.length);
+    version.addAll(value.split('.').map<int>(int.parse));
+  }
+
+  void increment(bool breaking) {
+    assert(version.length == 3);
+    if (breaking) {
+      version[1] += 1;
+      version[2] = 0;
+    } else {
+      version[2] += 1;
+    }
+    dirty = true;
+  }
+
+  void setMode(bool release) {
```

Comment:
The asymmetry that setMode's argument is true for release mode but the dev flag field is true for non-release mode seems a bit confusing.  It might be clearer to use an enum instead of bools for the mode.